samples/matter/lock/sample.yaml

```diff
@@ -2,27 +2,44 @@ sample:
   name: Matter Lock
   description: Matter Lock example
 tests:
-  matter.lock:
+  matter.lock.debug:
     build_only: true
     platform_allow: nrf52840dk_nrf52840 nrf5340dk_nrf5340_cpuapp nrf21540dk_nrf52840
     integration_platforms:
       - nrf52840dk_nrf52840
       - nrf5340dk_nrf5340_cpuapp
       - nrf21540dk_nrf52840
```

Comment:
Same here, not sure why we allow building DFU variants for nrf21540dk_nrf52840, but maybe this could be cleaned up.